kv/local_sender.go

```diff
@@ -123,6 +123,12 @@ func (ls *LocalSender) Send(ctx context.Context, call proto.Call) {
 	var err error
 	var store *storage.Store
 
+	defer func() {
+		if err != nil {
```

Comment:
This pattern seems fragile: if `err` is ever shadowed then the error wouldn't be returned correctly. I'd rather call `SetGoError` each time we have `if err != nil { return }` instead of calling it in a `defer`.